apisix/plugins/fault-injection.lua

```diff
@@ -77,11 +151,30 @@ end
 function _M.rewrite(conf, ctx)
     core.log.info("plugin rewrite phase, conf: ", core.json.delay_encode(conf))
 
-    if conf.delay and sample_hit(conf.delay.percentage) then
+    local err
+    local abort_vars = true
+    if conf.abort and conf.abort.vars then
+        abort_vars, err = vars_match(conf.abort.vars, ctx)
+        if err then
+            return 500, err
```

Comment:
Better to log the error.

Reply:
The log is recorded here: [https://github.com/apache/apisix/pull/3363/files#diff-a2a9ef17c5a2a556f996066b04132d8dd9a848b31eff4a340f5b154283af98deR90 ](https://github.com/apache/apisix/pull/3363/files#diff-a2a9ef17c5a2a556f996066b04132d8dd9a848b31eff4a340f5b154283af98deR90)